### PR TITLE
SDA-3565 - Allow C2 / extensions to determine the presence of Citrix media redirection

### PR DESCRIPTION
**Pull request by rwestberg**
## Description

In order to support Citrix Media Optimization, it is necessary for an extension to be able to query the current status. Only when the redirection support is active is it necessary to proceed with loading and initializing the full Citrix client-side SDK.

### CODE REVIEW

File: src/app/citrix-handler.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { enumerateValues, HKEY, RegistryValueType } from 'registry-js';
+
+export const getCitrixMediaRedirectionStatus = ():
+  | 'inactive'
```
Comment (sbenmoussati) [inline src/app/citrix-handler.ts:4]: you can define here an enum that would have 'inactive' 'supported' or 'unsupported' values.

File: src/app/citrix-handler.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  | 'supported'
+  | 'unsupported' => {
+  const values = enumerateValues(
+    HKEY.HKEY_CURRENT_USER,
+    'Software\\Citrix\\HDXMediaStream',
+  );
+
+  for (const value of values) {
+    if (value.name === 'MSTeamsRedirectionSupport') {
```
Comment (sbenmoussati) [inline src/app/citrix-handler.ts:13]: same here for 'MSTeamsRedirectionSupport'

File: src/renderer/ssf-api.ts [hunk 3]
```diff
@@ -730,6 +730,18 @@ export class SSFApi {
   public supportedSettings(): string[] {
     return SUPPORTED_SETTINGS || [];
   }
+
+  /**
+   * Retrieves the current status of Citrix' media redirection feature
+   * @returns status
+   */
+  public getCitrixMediaRedirectionStatus(): Promise<
+    'inactive' | 'supported' | 'unsupported'
```
Comment (sbenmoussati) [inline src/renderer/ssf-api.ts:739]: defined enum would be reused here

**Comment by sbenmoussati** [comment]
globally LGTM, few comments added.
[Question]: this implementation is specific to Windows. Do we need to do something similar for macOS ?

**Review by KiranNiranjan — APPROVED** [review]
LGTM 👍🏻  and agree with the changes requested by @sbenmoussati

File: src/app/citrix-handler.ts (hunk 1 above)
Comment (rwestberg) [reply src/app/citrix-handler.ts]: Makes sense, updated

**Comment by rwestberg** [comment]
> globally LGTM, few comments added. [Question]: this implementation is specific to Windows, correct? Do we need to do something similar for macOS ?

I don't think there's anything needed for other OSes - this is only supported when running the client on a Citrix server which only exists on Windows (you can run the Citrix client on macOS to interact with it, but the SDA code would still be running on the Windows server).

**Comment by sbenmoussati** [comment]
> > globally LGTM, few comments added. [Question]: this implementation is specific to Windows, correct? Do we need to do something similar for macOS ?
> 
> I don't think there's anything needed for other OSes - this is only supported when running the client on a Citrix server which only exists on Windows (you can run the Citrix client on macOS to interact with it, but the SDA code would still be running on the Windows server).

"Citrix server which only exists on Windows", that's the info I was missing, thanks for clarifying !

**Pull request merged**